lib/rubocop/cop/rspec/expect_change.rb

```diff
@@ -10,6 +10,15 @@ module RSpec
       #
       # This cop can be configured using the `EnforcedStyle` option.
       #
+      # @safety
+      #   Autocorrection is unsafe because `block` style evaluates the object
+      #   twice (before and after evaluating `expect` block), whereas
+      #   `method_call` style evaluates the object only once.
```

Comment:
What do you mean by “object” here? Statement or a series of statements inside the block?

Reply:
After looking at the original PR, it’s clear to me. Nice catch, and thanks for marking the cop, it’s totally justified.

Can you please provide a bit more context in safety docs to make it obvious for the reader?

Reply:
If I understand https://github.com/rubocop/rubocop-rspec/issues/1174 correctly, the issue is that the `method_call` style calls `message` twice on the same `receiver`, so to speak. But with the `block` style, the entire block is evaluated twice – including instantiating two different `receiver` objects.

In #1174, the `#time` method _is_ being called twice in each example, which confused me at first.

Reply:
> What do you mean by “object” here? Statement or a series of statements inside the block?

I followed the terminology used in the previous section ("Enforces either passing object and attribute as...") which I think was in turn borrowed from the example in [RSpec docs](https://rspec.info/features/3-13/rspec-expectations/built-in-matchers/change/):
```ruby
expect { do_something }.to change(object, :attribute)
```

Instead of object/attribute, more precise terminology would be receiver/message, but I'm not sure if that is more approachable?